### PR TITLE
Changes to further OSX compilation

### PR DESCRIPTION
**Pull request by jdekonin**
The changes to j9porterror.h were originally from @DanHeidinga that I took over as per his agreement in #36.  It needed a minor modification to compile on windows, an (I_32) addition.

### CODE REVIEW

File: runtime/port/sysvipc/j9shmem.c [hunk 1]
```diff
@@ -1613,12 +1613,16 @@ openSharedMemory (J9PortLibrary *portLibrary, intptr_t fd, const char *baseFile,
 				goto failDontUnlink;
 			}
 		} else {
-#if defined (__GNUC__) || defined (AIXPPC)
-#if defined (__GNUC__)
+#if defined(__GNUC__) || defined(AIXPPC)
+#if defined(__GNUC__) && !defined(OSX)
```
Comment (DanHeidinga) [inline runtime/port/sysvipc/j9shmem.c]: Can this be written as:
```
#if defined(OSX)
/*Use ._key for OSX*/
 +			if (buf.shm_perm._key != controlinfo->common.ftok_key)
#elif defined(__GNUC__)
			/*Use .__key for __GNUC__*/
  			if (buf.shm_perm.__key != controlinfo->common.ftok_key)		  			
#elif defined(AIXPPC) 
                        ......
#endif
```

File: runtime/port/sysvipc/j9shsem_deprecated.c [hunk 2]
```diff
@@ -1077,12 +1077,16 @@ openSemaphore(struct J9PortLibrary *portLibrary, intptr_t fd, char *baseFile, j9
 				goto failDontUnlink;
 			}
 		} else {
-#if defined (__GNUC__) || defined (AIXPPC) || defined(J9ZTPF)
-#if defined (__GNUC__) && !defined(J9ZTPF)
+#if defined(__GNUC__) || defined(AIXPPC) || defined(J9ZTPF)
+#if defined(__GNUC__) && !defined(J9ZTPF) && !defined(OSX)
```
Comment (DanHeidinga) [inline runtime/port/sysvipc/j9shsem_deprecated.c]: Can a similar refactoring as suggested above be done here as well?

**Comment by jdekonin** [comment]
Refactoring performed.

File: runtime/port/sysvipc/j9shsem_deprecated.c [hunk 3]
```diff
@@ -1077,18 +1077,19 @@ openSemaphore(struct J9PortLibrary *portLibrary, intptr_t fd, char *baseFile, j9
...
+#if defined(OSX)
+			/*Use _key for OSX*/
+			if (buf.sem_perm._key != controlinfo->ftok_key)
+#elif defined(AIXPPC)
 			/*Use .key for AIXPPC*/
 			if (buf.sem_perm.key != controlinfo->ftok_key)
-#endif
-#if defined(J9ZTPF)
+#elif defined(J9ZTPF)
             /*Use .key for z/TPF */
```
Comment (DanHeidinga) [inline runtime/port/sysvipc/j9shsem_deprecated.c]: Can you fix the comment spacing here since the files being modified anyway?

**Comment by DanHeidinga** [comment]
@jdekonin Change looks good to me and I'll launch builds on it after the minor comment indenting issue is fixed

File: runtime/port/sysvipc/j9shsem_deprecated.c (hunk 3 above)
Comment (jdekonin) [reply runtime/port/sysvipc/j9shsem_deprecated.c]: Fixed, sorry I should've caught that.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux

**Pull request merged**